### PR TITLE
Have git check out main branch if present

### PR DESCRIPTION
**Pull request by zzhengnan**
## Description

Modify git aliases to check out the `main` branch instead of `master` if the former is present.

## Motivation and Context

To reflect the growing adoption of `main` as the default branch, in place of `master`. 

Currently aliases that rely on `git checkout master` don't work in repos that have adopted this new pattern. With changes in this PR, the modified aliases will behave as follows
- If a `main` branch is present in the repo, check out the `main` branch
- If `master` is present, check out `master`
- If both branches are present, priority is given to `main`

## How Has This Been Tested?

I did the following
1. Create two repos, one using `master` as the default branch and another using `main`
2. Create a new branch `dev` in both repos, and check out this branch
3. Run `gcom` and confirm the expected default branch has been checked out in both repos

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] If my change requires a change to the documentation, I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] If I have added a new file, I also added it to ``clean_files.txt`` and formatted it using ``lint_clean_files.sh``.
- [ ] I have added tests to cover my changes, and all the new and existing tests pass.

### CODE REVIEW

File: aliases/available/git.aliases.bash [hunk 1]
```diff
@@ -196,3 +194,17 @@ esac
 function gdv() {
 	git diff --ignore-all-space "$@" | vim -R -
 }
+
+function gcom() {
+	if git branch | grep -q main; then
```
Comment (zzhengnan) [inline aliases/available/git.aliases.bash:199]: More context on the `-q` flag:
https://stackoverflow.com/questions/56170215/what-is-the-point-of-grep-q

Comment (zzhengnan) [reply aliases/available/git.aliases.bash:199]: grep's man page:
https://www.man7.org/linux/man-pages/man1/grep.1.html

**Review by NoahGorny — CHANGES_REQUESTED** [review]
Hey @zzhengnan, great idea!

There are many aliases in this file which use hard coded `master` string. Maybe create an helper function `get_default_branch` or something like that?

**Comment by zzhengnan** [comment]
@NoahGorny Thanks for the suggestion! Ready for review again. Here's how I tested the code.

Inside a repo whose default branch is `main`.
```
$ git branch
* dev
  main

$ get_default_branch
main

$ gcom
Switched to branch 'main'
```

Inside a repo whose default branch is `master`.
```
$ git branch
* dev
  master

$ get_default_branch
master

$ gcom
Switched to branch 'master'
```

Inside a repo where both `main` and `master` are present.
```
$ git branch
* dev
  main
  master

$ get_default_branch
main

$ gcom
Switched to branch 'main'
```

**Review by NoahGorny — APPROVED** [review]
Well done @zzhengnan !

**Comment by zzhengnan** [comment]
@NoahGorny Thanks for the review. Could you help merge the PR?

**Comment by NoahGorny** [comment]
> @NoahGorny Thanks for the review. Could you help merge the PR?

I usually wait around for some time to let other contributors chime in, and will merge the PR in the following days :smiley:

**Comment by zzhengnan** [comment]
Makes perfect sense. Thanks again.

**Pull request merged**